### PR TITLE
PD-4573 Lambda handler code

### PR DESCRIPTION
**Pull request by forjayssake**
Updated handler pseudo code to call road distance request handler
Rename provider to main
Main validates and now logs incoming (CCS) requests
Very basic logger config added to allow selecting logger instance by name with separate paths
Unit tests and associated make targets

### CODE REVIEW

File: application/tests/unit/test_roaddistance.py [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
-from application.provider import Provider
+from application.main import RoadDistance
 
 
-class TestProvider:
+class TestRoadDistance:
 
-    provider = Provider()
+    road_distance = RoadDistance({})
     json_path: str = "test_json/"
```
[the remark this answers is not in the document]
Comment (forjayssake) [reply application/tests/unit/test_roaddistance.py:9]: Fixing now

File: application/handler.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+def process_road_distance_request(event, context):
+    try:
+        road_distance = RoadDistance(json.loads(event))
+        status_code = road_distance.process_request()
+    except Exception as ex:
+        status_code = 500
+
+    response = {
+        "status_code": status_code,
```
Comment (terado) [inline application/handler.py:13]: Is this actually returning the status code in the header

File: application/rdlogger.py [hunk 3]
```diff
@@ -34,18 +34,29 @@
     Provider Response Success - Per returned service
     YYYY/MM/DD 00:00:00.000000+0100|info|lambda|<request_id>|<transaction_id>|roaddistancepilot|providerresponse|success|reference=<serviceUid>|unreachable=<yes/no>|distance=####
 """
+from configparser import ConfigParser
 import logging
+import json
```
Comment (terado) [inline application/rdlogger.py]: is this being used

File: application/tests/unit/test_handler.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        request = self.fetch_json("dos_road_distance_api_invalid_missing_element.json")
+        response = handler.process_road_distance_request(request, None)
+        assert response["status_code"] == 500
+        assert os.path.isfile(self.log_path) == False
+
+    def purge_test_logs(self):
+        if os.path.isfile(self.log_path):
+            os.remove(self.log_path)
+
+    def fetch_json(self, file_name: str) -> str:
```
Comment (terado) [inline application/tests/unit/test_handler.py:39]: I think we have this code in 3 places now - perhaps time for a helper class

Comment (forjayssake) [reply application/tests/unit/test_handler.py:39]: I thought about this, something we can add in on any ticket, but this one is doing something slightly different (not that it couldn't be coded around)

**Review by tamaragoldschmidt — APPROVED** [review]

**Pull request merged**